File: bcrypt-add.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import org.mindrot.jbcrypt.BCrypt;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

public class bcrypt-add {

    private static final String JSON_FILE = "bcrypt.json";

    public static void main(String[] args) {
        if (args.length < 3) {
            System.err.println("Usage: java bcrypt-add <username> <password> <rounds>");
            System.exit(1);
        }

        String username = args[0];
        String password = args[1];
        int rounds = Integer.parseInt(args[2]);

        // Generate salt for bcrypt
        String salt = BCrypt.gensalt(rounds);

        // Hash password using bcrypt
        String hashedPassword = BCrypt.hashpw(password, salt);

        JSONObject newUser = new JSONObject();
        newUser.put("username", username);
        newUser.put("password", hashedPassword);
        newUser.put("salt", salt);

        try {
            File file = new File(JSON_FILE);
            JSONArray usersArray;

            if (file.exists()) {
                // Read existing JSON file
                JSONParser parser = new JSONParser();
                Object obj = parser.parse(new FileReader(JSON_FILE));
                usersArray = (JSONArray) obj;
            } else {
                // Create new JSON file
                usersArray = new JSONArray();
            }

            // Add new user to JSON array
            usersArray.add(newUser);

            // Write JSON array to file
            FileWriter fileWriter = new FileWriter(JSON_FILE);
            fileWriter.write(usersArray.toJSONString());
            fileWriter.close();

            System.out.println("User '" + username + "' added to " + JSON_FILE);

        } catch (IOException | ParseException e) {
            e.printStackTrace();
        }
    }
}

File: bcrypt-add.py
import sys
import json
import bcrypt

JSON_FILE = "bcrypt.json"

def main():
    if len(sys.argv) < 4:
        print("Usage: python bcrypt-add.py <username> <password> <rounds>")
        sys.exit(1)

    username = sys.argv[1]
    password = sys.argv[2]
    rounds = int(sys.argv[3])

    # Generate salt for bcrypt
    salt = bcrypt.gensalt(rounds)

    # Hash password using bcrypt
    hashedPassword = bcrypt.hashpw(password.encode('utf-8'), salt)

    newUser = {
        "username": username,
        "password": hashedPassword.decode('utf-8'),
        "salt": salt.decode('utf-8')
    }

    try:
        with open(JSON_FILE, 'r') as file:
            usersArray = json.load(file)
    except FileNotFoundError:
        usersArray = []

    # Add new user to JSON array
    usersArray.append(newUser)

    try:
        with open(JSON_FILE, 'w') as file:
            json.dump(usersArray, file, indent=4)

        print(f"User '{username}' added to {JSON_FILE}")

    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    main()
